src/ProxyInterfaceSourceGenerator/Models/ProxyData.cs

```diff
@@ -4,13 +4,38 @@ namespace ProxyInterfaceSourceGenerator.Models;
 
 internal class ProxyData
 {
+    public ProxyData(string @namespace,
+                     string namespaceDot,
+                     string shortInterfaceName,
+                     string fullInterfaceName,
+                     string fullRawTypeName,
+                     string shortTypeName,
+                     string fullTypeName,
+                     List<string> usings,
+                     bool proxyBaseClasses,
+                     ProxyClassAccessibility accessibility)
+    {
+        Namespace = @namespace ?? throw new ArgumentNullException(nameof(@namespace));
+        NamespaceDot = namespaceDot ?? throw new ArgumentNullException(nameof(namespaceDot));
+        ShortInterfaceName = shortInterfaceName ?? throw new ArgumentNullException(nameof(shortInterfaceName));
+        FullInterfaceName = fullInterfaceName ?? throw new ArgumentNullException(nameof(fullInterfaceName));
+        FullRawTypeName = fullRawTypeName ?? throw new ArgumentNullException(nameof(fullRawTypeName));
+        ShortTypeName = shortTypeName ?? throw new ArgumentNullException(nameof(shortTypeName));
+        FullTypeName = fullTypeName ?? throw new ArgumentNullException(nameof(fullTypeName));
+        Usings = usings ?? throw new ArgumentNullException(nameof(usings));
+        ProxyBaseClasses = proxyBaseClasses;
+        Accessibility = accessibility;
+    }
+
     public string Namespace { get; init; }
```

Comment:
If only this new constructor is used, then all properties can be `{ get; }`, correct?

Reply:
You are absolutely correct. I stopped, when the warnings disappeared. Resharper probably would have picked that up. That's not a feature I use a lot.